### ISSUE TITLE
can't authenticate via attribute 'sAMAccountName',

### ISSUE THREAD

**Issue by neocy** [Kong/kong#3705]
@shashiranjan84 

LDAP worked well when use attribute 'cn'.
I have the same value of 'cn' and  'sAMAccountName'.
so I believe every config should be the same except attribute.
when I change attribute from 'cn' to 'sAMAccountName'.
authenticate failed.
I also try low case 'samaccountname'

- Kong version
0.14.0
- error logs:
2018/08/15 05:48:22 [error] 51929#0: *3579840 [lua] access.lua:81: 
  Error: The supplied credential is invalid.
  Details: 80090308: LdapErr: DSID-0C09042F, comment: AcceptSecurityContext error, data 52e, v2580, client: 10.192.242.44, server: kong, request: 
- Operating system
centos 7
- config
{
  "created_at": 1534237880000,
  "config": {
    "start_tls": false,
    "timeout": 10000,
    "base_dn": "OU=Users,OU=Dev,OU=Test,DC=test,DC=org",
    "hide_credentials": true,
    "ldap_host": "test.ldap.org",
    "ldap_port": 389,
    "cache_ttl": 60,
    "verify_ldap_host": false,
    "anonymous": "",
    "attribute": "cn",
    "header_type": "ldap",
    "keepalive": 60000
  },
  "id": "01ad7355-9080-4ffe-8fee-5c01dd63a671",
  "name": "ldap-auth",
  "enabled": true,
  "route_id": "c2a99645-17d5-425b-8860-8910cb11009f"
}

**Comment by littlechicks** [comment]
I have the same problème. Only CN works.. Is there an fix?

**Comment by littlechicks** [comment]
I have investigated. In fact, the plugin doesn't make difference on the authentication username. It always use a DN syntax. So even if we specify sAMAccoutName as the attribute, ldap-auth plugin will send:
sAMAccount=user, DC=domain, DC=com

It cannot be work in that way. LDAP sAMAccount must be the following syntax:
DOMAIN.COM\user
Or
user@domain.com

See my issue :
https://github.com/Kong/kong/issues/5121


I've proposed a fix in pull request :
https://github.com/Kong/kong/pull/5149
